modules/service/data-integrity/elliptic-data-integrity-service.js

```diff
@@ -0,0 +1,150 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+const elliptic = require('elliptic');
+
+// eslint-disable-next-line new-cap
+const secp256k1 = new elliptic.ec('secp256k1');
+const sha3 = require('js-sha3');
+
+const DataIntegrityService = require('./data-integrity-service');
+const BytesUtilities = require('../../BytesUtilities');
+
+// eslint-disable-next-line no-cond-assign,func-names,no-unsafe-finally,no-undef-init,no-undef
+const _slicedToArray = (function () { function sliceIterator(arr, i) { const _arr = []; let _n = true; let _d = false; let _e = undefined; try { for (let _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i.return) _i.return(); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } throw new TypeError('Invalid attempt to destructure non-iterable instance'); }; }());
+
+const SIGNATURE_TYPE = 'ethereum-signature';
+
+class EllipticDataIntegrityService extends DataIntegrityService {
+    sign(content, privateKey) {
+        super.sign(content, privateKey);
```

Comment:
Why do we call super here (and also in verify and recover) if the super function is empty?

Reply:
DataIntegrityService is an interface and for now has no purpose. However, it could be useful if each implementation has something in common which could be placed into a common class. If you find it unnecessary, we will remove it.